src/robotFactory.js

```diff
@@ -1,10 +1,70 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
```

Comment:
It is better to fill object in place rather than by separate props assignments.
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```